app/models/ems_refresh/save_inventory_middleware.rb

```diff
@@ -57,9 +57,17 @@ def save_middleware_servers_inventory(ems, hashes, target = nil)
               end
 
     hashes.each do |h|
-      nativeid = h.fetch_path(:middleware_server_group, :nativeid)
-      h[:server_group_id] = MiddlewareServerGroup.find_by_nativeid(nativeid).try(:id) unless nativeid.nil?
+      server_group_id = h.fetch_path(:middleware_server_group, :id)
+      unless server_group_id
```

Comment:
Yes, that was my intention. I can rewrite it to `if server_group_id.nil?` , it's more readable